frontend/packages/console-shared/src/components/formik-fields/SwitchField.tsx

```diff
@@ -1,21 +1,23 @@
-/* eslint-disable no-unused-vars, no-undef */
 import * as React from 'react';
 import { useField } from 'formik';
 import { FormGroup, Switch } from '@patternfly/react-core';
-import { getFieldId } from '@console/dev-console/src/components/formik-fields/field-utils';
-import { CheckboxFieldProps } from '@console/dev-console/src/components/formik-fields/field-types';
+import { getCheckboxFieldId } from './field-utils';
+import { CheckboxFieldProps } from './field-types';
 
 const SwitchField: React.FC<CheckboxFieldProps> = ({
```

Comment:
Yikes, had no idea `SwitchField` was so similar to `CheckboxField`... we need to deduplicate this code using some sort of HOC or something. The logic of both of these components is near identical.

@christianvogt @rohitkrai03 I recommend we look to clean this up after this PR as to keep everything from crossing wires.

Reply:
Interesting, this didn't come from `dev-console`. I just assumed all fields were in our package. Either way, this should probably be cleaned up, it's near identical code.

Reply:
@andrewballantyne I think we can merge both of these components and control the component it renders internally using a prop. Something like 
```js
<CheckboxField as="switch" {...prop} />
```
This is inline with how formik handles rendering form fields with different components. We could make `checkbox` as default variant as well.

Reply:
I like keeping `SwitchField` as a separate component as it is easier for devs to know it exists. How about this? https://github.com/jtomasek/console/commit/9f2a9762358b103b5a2c6a5e96011dba827f3f4f
I've used render props pattern and refactored common code into `CheckBoxFieldBase` component. It would be IMHO better to keep that change in a separate PR.

Reply:
Definitely agree, it should be another PR, this one should track the refactor (it's a big enough change to be done on it's own). 

FWIW, I lean the render prop way that @jtomasek suggested. I had envisioned more of a function composition where the _Component_ would be passed in and invoked with JSX inside a nested function -- but it seems our hook lint doesn't like that idea. The render prop is a decent alternative imo.